javascript/modules/client/src/client.ts

```diff
@@ -579,50 +593,152 @@ export class Client extends ClientCore implements IClient {
-  private _getDao(daoAddressOrEns: string): Promise<DaoDetails> {
-    // TODO: Implement actual fetch logic using subgraph.
-
-    if (!daoAddressOrEns) {
-      throw new Error("Invalid DAO address or ENS");
+  private async _getDao(daoAddressOrEns: string): Promise<DaoDetails | null> {
+    if (!isAddress(daoAddressOrEns)) {
+      // TODO
+      // ens resolve
+      throw new Error("Invalid address");
+    }
+    try {
+      await this.graphql.ensureOnline();
+      const client = this.graphql.getClient();
+      const { dao } = await client.request(QueryDao, {
+        address: daoAddressOrEns,
+      });
+      if (!dao) {
+        return null;
+      }
+      // const stringMetadata = await this.ipfs.fetchString(dao.metadata);
+      // TODO
+      // this is a temporal fix and should be changed by the line above
+      // but the current daos in subgraph dont have a proper metadata
+      const stringMetadata = await this.ipfs.fetchString(
+        "QmebY8BGzWBUyVqZFMaFkFmz3JrfaDoFP5orDqzJ1uiEkr",
+      );
+      const metadata = JSON.parse(stringMetadata);
+      return {
+        address: dao.id,
+        ensDomain: dao.name,
+        metadata: {
+          name: metadata.name,
+          description: metadata.description,
+          avatar: metadata.avatar || null,
+          links: metadata.links,
+        },
+        creationDate: new Date(parseInt(dao.createdAt) * 1000),
+        // TODO update when new subgraph schema is deployed
+        plugins: dao.packages.map(
+          (
+            pkg: { pkg: { id: string; __typename: string } },
+          ): InstalledPluginListItem => {
+            return {
+              instanceAddress: pkg.pkg.id,
+              id: pkg.pkg.__typename,
+              version: "0.0.1",
+            };
+          },
+        ),
+      };
+    } catch (err) {
+      throw new Error("Cannot fetch the dao details data from GraphQL");
     }
-    return new Promise((resolve) => setTimeout(resolve, 1000)).then(() =>
-      getDummyDao(daoAddressOrEns)
-    );
   }
 
-  private _getDaos({
+  private async _getDaos({
     limit = 10,
-    // TODO
-    // uncomment this
-    // skip = 0,
-    // direction = SortDirection.ASC,
-    // sortBy = DaoSortBy.CREATED_AT
-  }: IDaoQueryParams): Promise<DaoDetails[]> {
-    const metadataMany: DaoDetails[] = [];
-    for (let index = 0; index < limit; index++) {
-      metadataMany.push(getDummyDao());
+    skip = 0,
+    direction = SortDirection.ASC,
+    sortBy = DaoSortBy.CREATED_AT,
+  }: IDaoQueryParams): Promise<DaoListItem[]> {
+    const daosDetails: DaoListItem[] = [];
+    try {
+      await this.graphql.ensureOnline();
+      const client = this.graphql.getClient();
+      const { daos } = await client.request(QueryDaos, {
+        limit,
+        skip,
+        direction,
+        sortBy,
+      });
+      for (let i = 0; i < daos.length; i++) {
+        const dao = daos[i];
+        // const stringMetadata = await this.ipfs.fetchString(dao.metadata);
```

Comment:
Handling the array in serial mode is going to be quite slow. Could we turn this into a `Promise.all(daos.map(dao => {...}))` operation?